src/acdl/acdl_decision_heuristics_equality.cpp

```diff
@@ -115,7 +119,10 @@ acdl_domaint::meet_irreduciblet acdl_decision_heuristics_equalityt::operator()
  std::vector<bool> non_cond_marked(non_conds_ns.size(), true);
  
  if(conds_ns.size() > 0 || non_conds_ns.size() > 0) {
-   while(!decision) {
+   assert(bad_decisions.size() == 0);
+   // bad decisions can either be repeatative 
+   // decisions or decision which are subsumed
+   while(!decision && bad_decisions.size()<1) {
```

Comment:
Bad decision here means "repetitive decision". This is just used for debugging to check whether we run into scenarios which returns repetitive decisions.